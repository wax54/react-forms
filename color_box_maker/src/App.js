import React from "react";
import BoxList from './BoxList';
import "./App.css";

/** Simple app that makes a box list which can make colored boxes of various sizes */

function App() {
  return (
    <div className="App">
      <BoxList />
    </div>
  );
}

export default App;
