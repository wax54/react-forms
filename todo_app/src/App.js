import React from "react";
import TodoList from './TodoList';
import "./App.css";

/** Simple app that shows a Todo List which can CRUD Todos. */

function App() {
  return (
    <div className="App">
      <TodoList />
    </div>
  );
}

export default App;
