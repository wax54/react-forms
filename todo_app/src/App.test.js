import React from 'react';
import { render, fireEvent} from "@testing-library/react";
import App from './App';

it("Doesn't Blow up", () => {
    render(<App />);
});

it('should match previous snapshot', function () {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
});

it("should add a new todo to the DOM", () => {
    let testText = "NEW TODO";
    const { getByLabelText, getByText} = render(<App />);

    const taskInput = getByLabelText("Task", { exact: false });

    const btn = getByText("Add Todo", { exact: false });


    fireEvent.change(taskInput, { target: { value: testText } });
    fireEvent.click(btn);

    expect(getByText(testText)).toBeInTheDocument();
});

it("should remove a TODO from the DOM when it's X is clicked", () => {
    let boxStats = {};

    const { getByLabelText, getByText, queryByText } = render(<App />);
    //setup START
    let testText = "NEW TODO";
    const taskInput = getByLabelText("Task", { exact: false });
    const btn = getByText("Add Todo", { exact: false });
    fireEvent.change(taskInput, { target: { value: testText } });
    fireEvent.click(btn);
    expect(getByText(testText)).toBeInTheDocument();
    //Setup END

    const deleteTodoBtn = getByText("X");
    fireEvent.click(deleteTodoBtn);

    expect(queryByText(testText)).not.toBeInTheDocument();
})
