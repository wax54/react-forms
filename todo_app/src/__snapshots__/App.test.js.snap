// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match previous snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="TodoList"
    >
      <form
        class="NewTodoForm"
      >
        <label
          for="title-input"
        >
          Task: 
        </label>
        <input
          id="title-input"
          name="title"
          placeholder="Wash Dog"
          required=""
          type="text"
          value=""
        />
        <button>
          Add Todo
        </button>
      </form>
      <ul
        class="TodoList-list"
      />
    </div>
  </div>
</DocumentFragment>
`;
